lib/components/src/Accordion/AccordionHeader.tsx

```diff
@@ -0,0 +1,199 @@
+import { styled } from '@storybook/theming';
+import React, { FC, useContext, useEffect, useRef, useState } from 'react';
+import { Icons } from '../icon/icon';
+import { AccordionItemContext } from './AccordionItemContext';
+
+export type AccordionHeaderProps = {
+  label?: string;
+  hideIcon?: boolean;
+  Icon?: React.ReactNode;
+  LabelProps?: React.HTMLAttributes<HTMLDivElement>;
+  onOpen?: () => {};
+  onClose?: () => {};
+  open?: boolean;
+  preventToggle?: boolean;
+} & React.HTMLAttributes<HTMLButtonElement>;
+
+export const AccordionHeader: FC<AccordionHeaderProps> = ({
```

Comment:
Does the AccorionHeader really need any logic at all? Looks like this is far more complex then it needs to be, but perhaps I'm missing something.
Why is this expanding and collapsing code here?

I feel as if, this component should simply be given a function to call, that's all. It's it's there, it's called.
I propose to simplify this to accept: `isOpen: boolean` & `onClick: Function`
If `onClick` is missing, the element isn't clickable.

Reply:
It could probably manage without it, but by making it able to control itself the responsibility for the logic behind that is placed in the file where it happens, which is a lot more readable for me when looking at it than having to then backtrack a prop to the outside to see what it does in another file.